src/services/dispatchers.ts

```diff
@@ -13,15 +13,19 @@ export async function sync(sdk: OAuth1Client, path: string) {
   const rql = rqlBuilder().eq('tags', cliManagedTag).build();
   const exhDispatchers = await dispatcherRepository.findAll(sdk, rql);
 
+  // Ensure all Dispatchers and actions have name fields
   for (const localDispatcher of localDispatchers) {
-    console.group(blue(`Synchronizing Dispatcher: ${localDispatcher.name}`));
-
-    // Ensure all Dispatchers and actions have name fields
     assertRequiredFields(localDispatcher);
```

Comment:
To me it makes more sense if we first check the file and don't sync anything if there is something invalid in it. 
Also the output is a bit less cluttered this way because syncing and validation are not entangled anymore. 

I prefer those advantages over the bit of performance optimalization.

Reply:
This is also an error on my side, we agreed that any validation fail is a full synchronization fail. It should of been how you declared it from the start.